taskboot/utils.py

```diff
@@ -109,15 +110,20 @@ def download_artifact(queue, task_id, artifact_name):
     except taskcluster.exceptions.TaskclusterAuthFailure:
         url = queue.buildUrl("getLatestArtifact", task_id, artifact_name)
 
-    # Download the artifact in a temporary file
-    _, ext = os.path.splitext(artifact_name)
-    _, path = tempfile.mkstemp(suffix="-taskboot{}".format(ext))
+    if output_directory is None:
+        # Download the artifact in a temporary file
+        _, ext = os.path.splitext(artifact_name)
+        _, path = tempfile.mkstemp(suffix="-taskboot{}".format(ext))
+    else:
+        # Download the artifact in a specific directory
+        path = output_directory + artifact_name
```

Comment:
Use `os.path.join` instead of string concatenation

Reply:
`output_directory` is a `pathlib.Path`, perhaps it would be better to use the `/` operator. What do you think?

Reply:
WFM